### PR TITLE
Handle Too Many Requests response

### PR DESCRIPTION
**Pull request by goozler**
Hello! Thank you for creating the gem!

One thing brings about some hacks to my main project because of too many requests issue. Would you mind to add an error handling for that case? It's suppressed now and it's hard to manage the issue outside of the gem.

### CODE REVIEW

File: spec/mailerlite/error_spec.rb [hunk 1]
```diff
@@ -23,6 +23,12 @@
         .to raise_error(MailerLite::NotFound, "Can't find requested items")
     end
 
+    it 'has too many requests error raised when status is 429' do
+      response = double(status: 429, body: [])
+      expect { raise MailerLite::Error.from_response(response) }
+        .to raise_error(MailerLite::TooManyRequests, "Too Many Requests")
```
[the remark this answers is not in the document]
Comment (goozler) [reply spec/mailerlite/error_spec.rb]: I kept double-quoted here because other specs have it in that file.

**Comment by jpalumickas** [comment]
Hello!

Thanks for Pull request! Looks good to me 😉

**Comment by jpalumickas** [comment]
Released v1.4.0 with this pull request

**Comment by goozler** [comment]
Really fast! Thank you very much, I appreciate it! ;-)

**Pull request merged**